Lib/test/test_asyncio/test_streams.py

```diff
@@ -1095,23 +1161,48 @@ def test_eof_feed_when_closing_writer(self):
 
     def test_stream_reader_create_warning(self):
         with self.assertWarns(DeprecationWarning):
-            asyncio.StreamReader(loop=self.loop)
-
-    def test_stream_reader_protocol_create_warning(self):
-        reader = asyncio.StreamReader(loop=self.loop,
-                                      _asyncio_internal=True)
-        with self.assertWarns(DeprecationWarning):
-            asyncio.StreamReaderProtocol(reader, loop=self.loop)
+            asyncio.StreamReader
 
     def test_stream_writer_create_warning(self):
-        reader = asyncio.StreamReader(loop=self.loop,
-                                      _asyncio_internal=True)
-        proto = asyncio.StreamReaderProtocol(reader, loop=self.loop,
-                                             _asyncio_internal=True)
         with self.assertWarns(DeprecationWarning):
-            asyncio.StreamWriter('transport', proto, reader, self.loop)
-
-
+            asyncio.StreamWriter
+
+    def test_stream_reader_forbidden_ops(self):
+        async def inner():
+            stream = asyncio.Stream(mode=asyncio.StreamMode.READ)
+            with self.assertRaisesRegex(RuntimeError, "The stream is read-only"):
+                await stream.write(b'data')
+            with self.assertRaisesRegex(RuntimeError, "The stream is read-only"):
+                await stream.writelines([b'data', b'other'])
+            with self.assertRaisesRegex(RuntimeError, "The stream is read-only"):
+                stream.write_eof()
+            with self.assertRaisesRegex(RuntimeError, "The stream is read-only"):
+                await stream.drain()
+
+        self.loop.run_until_complete(inner())
+
+    def test_stream_writer_forbidden_ops(self):
+        async def inner():
+            stream = asyncio.Stream(mode=asyncio.StreamMode.WRITE)
```

Comment:
`DeprecationWarning` over absence of `_asyncio_internal=True`

Reply:
Fixed